### PR TITLE
chore: drop q where possible

### PR DESCRIPTION
**Pull request by erisu**
### Motivation, Context & Description

- Replace `q` module with native promises https://github.com/apache/cordova/issues/7
- Closes #76

`superspawn` is not removed and still uses `Q` for `progress` & `notify`.

### Testing

- `npm t`

### Checklist

- [x] I've run the tests to see all new and existing tests pass

### CODE REVIEW

File: src/PluginManager.js [hunk 1]
```diff
@@ -80,9 +79,9 @@ class PluginManager {
      * @returns {Promise} Returns a Q promise, either resolved in case of success, rejected otherwise.
```
Comment (QuentinFarizon) [inline src/PluginManager.js:79]: `Returns a Promise`

Comment (timbru31) [reply src/PluginManager.js:79]:
```suggestion
     * @returns {Promise} Returns a Promise, either resolved in case of success, rejected otherwise.
```

**Review by raphinesse — APPROVED** [review]

**Comment by erisu** [comment]
@raphinesse I rebased this PR since it has not been updated in the last 3 years.

As of now, I can not find `Q` being used in any of the higher packages. It only remains in development repos. E.g. coho, paramedic, etc. 

This is the only place left in this repo where `Q` is still being used.

https://github.com/erisu/cordova-common/blob/v4-drop-q/src/superspawn.js#L23

I am guessing I left it because of the usage **[`d.notify()`](https://github.com/erisu/cordova-common/blob/v4-drop-q/src/superspawn.js#L115)**.

I don't recall any options to get around this and I dont remember what happens higher up the chain...

**Comment by erisu** [comment]
Also, this repo is currently setup for taking in next major release PRs. (`5.0.0`)

Any ideas or suggestions on what to do with this PR?

**Comment by erisu** [comment]
Spawn is used in these locations:

* **`cordova-windows`** which is a deprecated platform and archived repo
* **`cordova-fetch`**
  * **[installPackage](https://github.com/apache/cordova-fetch/blob/master/index.js#L87)**
    * Method is called [here](https://github.com/apache/cordova-fetch/blob/master/index.js#L69)
  * **[uninstall](https://github.com/apache/cordova-fetch/blob/master/index.js#L165)**
    * Method is called lib's [platform remove](https://github.com/apache/cordova-lib/blob/master/src/cordova/platform/remove.js#L89) and [plugman remove](https://github.com/apache/cordova-lib/blob/master/src/plugman/uninstall.js#L129).
* And also in one test spec file for this repo.

**Review by breautek — APPROVED** [review]

**Pull request merged**